core/src/banking_stage.rs

```diff
@@ -269,17 +269,15 @@ impl BankingStage {
             bank.load_and_execute_transactions(txs, lock_results, MAX_RECENT_BLOCKHASHES / 2);
         let load_execute_time = now.elapsed();
 
-        let record_time = {
-            let now = Instant::now();
-            Self::record_transactions(bank.slot(), txs, &results, poh)?;
```

Comment:
Ouch haha

Reply:
It might be good to move the logic before the unlock into a separate function with a result which we can handle here, so nobody ever makes a similar mistake again?

Reply:
Moved the logic to separate function.

Reply:
Well. Would be best to make an object that wraps the locks, on drop it unlocks the accounts and also add a test. We can file an issue for those things though.

Reply:
Opened a separate issue to track wrapping this in an object
https://github.com/solana-labs/solana/issues/3463